physical/consul.go

```diff
@@ -84,14 +160,245 @@ func newConsulBackend(conf map[string]string) (Backend, error) {
 
 	// Setup the backend
 	c := &ConsulBackend{
-		path:       path,
-		client:     client,
-		kv:         client.KV(),
-		permitPool: NewPermitPool(maxParInt),
+		path:                path,
+		client:              client,
+		kv:                  client.KV(),
+		permitPool:          NewPermitPool(maxParInt),
+		consulClientConf:    consulConf,
+		serviceName:         service,
+		checkTimeout:        checkTimeout,
+		checkTimer:          time.NewTimer(checkTimeout),
+		disableRegistration: disableRegistration,
 	}
 	return c, nil
 }
 
+// UpdateAdvertiseAddr provides a pre-initialization hook for updating
+// Consul's advertise address.
+func (c *ConsulBackend) UpdateAdvertiseAddr(addr string) error {
+	if c.running {
+		return fmt.Errorf("service registration unable to update advertise address, backend already running")
+	}
+
+	host, port, err := parseAdvertiseAddr(addr)
+	if err != nil {
+		return errwrap.Wrapf(fmt.Sprintf(`failed to parse advertise address "%v": {{err}}`, addr), err)
+	}
+
+	c.advertiseHost = host
+	c.advertisePort = int(port)
+	return nil
+}
+
+// serviceTags returns all of the relevant tags for Consul.
+func serviceTags(active bool) []string {
+	activeTag := "standby"
+	if active {
+		activeTag = "active"
+	}
+	return []string{activeTag}
+}
+
+func (c *ConsulBackend) AdvertiseActive(active bool) error {
+	c.serviceLock.Lock()
+	defer c.serviceLock.Unlock()
+
+	// Vault is still bootstrapping
+	if c.service == nil {
+		return nil
+	}
+
+	// Save a cached copy of the active state: no way to query Core
+	c.active = active
+
+	// Ensure serial registration to the Consul agent.  Allow for
+	// concurrent calls to update active status while a single task
+	// attempts, until successful, to update the Consul Agent.
+	if !c.disableRegistration && atomic.CompareAndSwapInt64(&c.registrationLock, 0, 1) {
+		defer atomic.CompareAndSwapInt64(&c.registrationLock, 1, 0)
+
+		// Retry agent registration until successful
+	registration_complete:
```

Comment:
This label seems unnecessary, as the only break statement in the for loop breaks straight out of it